### PR TITLE
Testing of various C++20 features

### PR DESCRIPTION
**Pull request by robertlipe**
A little bit of everything just to see how well C++20 is supported. This is just
a few commits for each major feature. If there's anything useful in this PR, I expect
to have to pick it apart and cherry pick as I can't imagine this whole PR going in as
one. 

See related discussion on #1204.

- Templatize FahrenheitToCelsius and inverse.
- Toe dip for C++-20: use host endianness.
- unicsv.cc
- Babys's first concept.
- Use enum class and C++20 'using' directive in xcsv.
- use using enum in units to reduce chatter
- DANGER: wreckless use of std::format for testing

### CODE REVIEW

File: fmt.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  using fmt::print;
+  using fmt::println;
+
+  using fmt::vformat;
+  using fmt::vformat_to;
+} ;
+
+// It's present everywhere else, right?
+#elif __has_include(<format.h>)
```
Comment (tsteven4) [inline fmt.h]: The standard header is `<format>`, not `<format.h>`.
https://en.cppreference.com/w/cpp/header/format

**Comment by tsteven4** [comment]
You had a bad merge on xcsv.cc.  I think this is what you meant:
[xcsv.zip](https://github.com/GPSBabel/gpsbabel/files/13250628/xcsv.zip)

File: fmt.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  using fmt::print;
+  using fmt::println;
+
+  using fmt::vformat;
+  using fmt::vformat_to;
+} ;
+
+// Fallback to the system version.
+#elif __has_include(<format>)
+# include <format.h>
```
Comment (tsteven4) [inline fmt.h:49]: <format>

Comment (tsteven4) [inline fmt.h:24]: Again, the system version is`<format>`

**Comment by tsteven4** [comment]
With the xcsv.cc I attached above, and actually including `<format>` after testing for it, on visual studio I get down to 7 errors, all in the MS supplied format include.
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2338	static_assert failed: 'Cannot format an argument. To make type T formattable, provide a formatter<T> specialization. See N4928 [format.arg.store]/2 and [formatter.requirements].'	C:\Users\tstev\Source\Repos\upstream\out\build\x64-Debug\upstream	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\format	3648	
Error	C2665	'std::_Format_arg_traits<_Context>::_Phony_basic_format_arg_constructor': no overloaded function could convert all the argument types	C:\Users\tstev\Source\Repos\upstream\out\build\x64-Debug\upstream	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\format	1873	
Error	C2993	'unknown-type': is not a valid type for non-type template parameter '_Test'	C:\Users\tstev\Source\Repos\upstream\out\build\x64-Debug\upstream	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\format	3430	
Error	C2641	cannot deduce template arguments for 'std::formatter'	C:\Users\tstev\Source\Repos\upstream\out\build\x64-Debug\upstream	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\format	3432	
Error	C2783	'std::formatter<_Ty,_CharT> std::formatter(void)': could not deduce template argument for '_Ty'	C:\Users\tstev\Source\Repos\upstream\out\build\x64-Debug\upstream	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\format	3432	
Error	C2780	'std::formatter<_Ty,_CharT> std::formatter(std::formatter<_Ty,_CharT>)': expects 1 arguments - 0 provided	C:\Users\tstev\Source\Repos\upstream\out\build\x64-Debug\upstream	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\format	3432	
Error	C2039	'parse': is not a member of 'std::formatter'	C:\Users\tstev\Source\Repos\upstream\out\build\x64-Debug\upstream	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\format	3433	
```

**Comment by robertlipe** [comment]
Actually, after I finally got through the merge noise in all this and could whwat was outstatnding, there's nothing to see here.

c++20, it is. Case closed.

**Pull request closed without merging**